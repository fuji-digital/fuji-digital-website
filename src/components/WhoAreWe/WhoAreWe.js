import { useWindowSize } from "@uidotdev/usehooks";
import { motion, useScroll, useTransform } from "framer-motion";

import {
  StyledBody,
  StyledContentContainer,
  StyledHeader,
  StyledInformationContainer,
  StyledIconContainer,
  StyledStaffPhoto,
  StyledAboutMe,
  CardContainer,
} from "./WhoAreWe.styles";
import { Card } from "../Card/Card";
import StaffPhotoEmily from "../../images/staff-photos/Emily.png";
import StaffPhotoGreg from "../../images/staff-photos/Greg.png";

export const WhoAreWe = () => {
  const { width } = useWindowSize();

  const { scrollYProgress } = useScroll();

  const fadeInOnScrollEmily = useTransform(
    scrollYProgress,
    [0.4, 0.75],
    [0, 1]
  ); // Fade in

  const fadeInOnScrollXEmily = useTransform(
    scrollYProgress,
    [0.4, 0.75],
    [100, 0]
  ); // Slide up

  const fadeInOnScrollGreg = useTransform(scrollYProgress, [0.4, 0.85], [0, 1]); // Fade in

  const fadeInOnScrollXGreg = useTransform(
    scrollYProgress,
    [0.4, 0.85],
    [100, 0]
  );

  return (
    <StyledBody id="who-are-we">
      <Card>
        <StyledHeader>Who Are We?</StyledHeader>
        <CardContainer>
          <motion.div
            style={{ opacity: fadeInOnScrollEmily, x: fadeInOnScrollXEmily }}
          >
            <StyledInformationContainer>
              <StyledContentContainer>
                <StyledStaffPhoto
                  src={StaffPhotoEmily}
                  style={{ border: "5px solid #5E17EB" }}
                ></StyledStaffPhoto>
                <h3>Emily Buckley</h3> Senior Engineer
              </StyledContentContainer>
              <StyledIconContainer
                color="#fff"
                gitHubUrl="https://github.com/emilybuc"
                linkedInUrl="https://www.linkedin.com/in/emily-buckley-a91a7118a/"
              />
              {width >= 550 && (
                <StyledAboutMe>
                  Hey there! I'm Emily, ðŸ˜Š an experienced Software Engineer,
                  boasting a solid 5+ years track record at LEGO and RS
                  Components. My passion lies in tackling challenges head-on and
                  crafting innovative solutions that make a difference. Beyond
                  work, I love to unwind by immersing myself in the captivating
                  world of board games, keeping fit at the gym, and playing with
                  my cat Hugo ðŸ˜º
                </StyledAboutMe>
              )}
            </StyledInformationContainer>
          </motion.div>
          <motion.div
            style={{ opacity: fadeInOnScrollGreg, x: fadeInOnScrollXGreg }}
          >
            <StyledInformationContainer>
              <StyledContentContainer>
                <StyledStaffPhoto
                  src={StaffPhotoGreg}
                  style={{ border: "5px solid #F1A1E7" }}
                ></StyledStaffPhoto>
                <h3>Gregory Carter</h3> Junior Engineer
              </StyledContentContainer>
              <StyledIconContainer
                color="#fff"
                gitHubUrl="https://github.com/Gregamoo"
                linkedInUrl="https://www.linkedin.com/in/gregamo/"
              />
              {width >= 550 && (
                <StyledAboutMe>
                  Hey, I'm Greg! ðŸ˜„ I trained as a video game designer but
                  pursued a career in software engineering, where I thrive on
                  solving complex technical challenges. My game design
                  background fuels my creativity and problem-solving skills in
                  the tech world. Outside of work, I love spending time with my
                  daughter, attending live concerts, playing video games, and
                  reading manga. I also enjoy going to the gym when I have time.
                  This blend of technical expertise and diverse interests keeps
                  me inspired and engaged in all aspects of my life.ðŸŒŸ
                </StyledAboutMe>
              )}
            </StyledInformationContainer>
          </motion.div>
        </CardContainer>
      </Card>
    </StyledBody>
  );
};
