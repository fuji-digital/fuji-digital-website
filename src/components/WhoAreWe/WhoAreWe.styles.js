import styled from "styled-components";
import { IconContainer } from "../SocialIcons/IconContainer";

export const StyledBody = styled.div`
  background: #13072e;
  font-family: "Poppins", sans-serif;
  padding: 1rem;
  color: #ffffff;
  h3 {
    margin: 0;
  }
`;

export const CardContainer = styled.div`
  @media screen and (min-width: 1200px) {
    display: flex;
    gap: 10px;
  }
`;
export const StyledHeader = styled.div`
  font-weight: bold;
  font-size: 1.5rem;
`;

export const StyledInformationContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  @media screen and (min-width: 768px) {
    justify-content: flex-start;
  }

  @media screen and (min-width: 1200px) {
    flex-basis: 50%;
  }
`;

export const StyledContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
`;

export const StyledStaffPhoto = styled.img`
  border-top-left-radius: 50% 50%;
  border-top-right-radius: 50% 50%;
  border-bottom-right-radius: 50% 50%;
  border-bottom-left-radius: 50% 50%;
  height: 8rem;
  width: 8rem;
`;

export const StyledIconContainer = styled(IconContainer)`
  flex-direction: column;
  align-self: end;
`;

export const StyledAboutMe = styled.p`
  align-self: flex-start;
`;
