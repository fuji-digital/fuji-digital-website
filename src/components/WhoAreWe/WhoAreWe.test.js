import renderer from "react-test-renderer";
import { WhoAreWe } from "./WhoAreWe";
import { render, screen } from "@testing-library/react";
import { useWindowSize } from "@uidotdev/usehooks";

jest.mock(`@uidotdev/usehooks`, () => ({
  useWindowSize: jest.fn(),
}));

jest.mock(`framer-motion`, () => ({
  ...jest.requireActual("framer-motion"),
  motion: {
    div: jest.fn().mockImplementation(({ children }) => <div>{children}</div>),
  },
}));

describe("WhoAreWe", () => {
  it("about me renders at 550px screen width", () => {
    useWindowSize.mockReturnValue({ width: 550 });

    render(<WhoAreWe />);
    const aboutMeEmily = screen.getByText(
      "Hey there! I'm Emily, ðŸ˜Š an experienced Software Engineer, boasting a solid 5+ years track record at LEGO and RS Components. My passion lies in tackling challenges head-on and crafting innovative solutions that make a difference. Beyond work, I love to unwind by immersing myself in the captivating world of board games, keeping fit at the gym, and playing with my cat Hugo ðŸ˜º"
    );
    expect(aboutMeEmily).toBeInTheDocument();

    const aboutMeGreg = screen.getByText(
      "Hey, I'm Greg! ðŸ˜„ I trained as a video game designer but pursued a career in software engineering, where I thrive on solving complex technical challenges. My game design background fuels my creativity and problem-solving skills in the tech world. Outside of work, I love spending time with my daughter, attending live concerts, playing video games, and reading manga. I also enjoy going to the gym when I have time. This blend of technical expertise and diverse interests keeps me inspired and engaged in all aspects of my life.ðŸŒŸ"
    );
    expect(aboutMeGreg).toBeInTheDocument();
  });

  it("about me does not render below 550px screen width", () => {
    useWindowSize.mockReturnValue({ width: 540 });

    render(<WhoAreWe />);
    const aboutMeEmily = screen.queryByText(
      "Hey there! I'm Emily, ðŸ˜Š an experienced Software Engineer, boasting a solid 5+ years track record at LEGO and RS Components. My passion lies in tackling challenges head-on and crafting innovative solutions that make a difference. Beyond work, I love to unwind by immersing myself in the captivating world of board games, keeping fit at the gym, and playing with my cat Hugo ðŸ˜º"
    );
    expect(aboutMeEmily).not.toBeInTheDocument();

    const aboutMeGreg = screen.queryByText(
      "Hey, I'm Greg! ðŸ˜„ I trained as a video game designer but pursued a career in software engineering, where I thrive on solving complex technical challenges. My game design background fuels my creativity and problem-solving skills in the tech world. Outside of work, I love spending time with my daughter, attending live concerts, playing video games, and reading manga. I also enjoy going to the gym when I have time. This blend of technical expertise and diverse interests keeps me inspired and engaged in all aspects of my life.ðŸŒŸ"
    );
    expect(aboutMeGreg).not.toBeInTheDocument();
  });

  it("matches snapshot", () => {
    useWindowSize.mockReturnValue({ width: 550 });

    const component = renderer.create(<WhoAreWe />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
