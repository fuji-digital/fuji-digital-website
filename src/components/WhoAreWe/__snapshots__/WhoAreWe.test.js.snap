// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhoAreWe matches snapshot 1`] = `
<div
  className="sc-gEvDqW bozaXs"
  id="who-are-we"
>
  <div
    className="sc-jXbVAB bqnbMq"
  >
    <div
      className="sc-fqkwJk jWucyp"
    >
      Who Are We?
    </div>
    <div
      className="sc-eqUzNf fzDmPm"
    >
      <div>
        <div
          className="sc-dcJtft eJDELm"
        >
          <div
            className="sc-iGgVNO ffHzaU"
          >
            <img
              alt="A staff photo of Emily"
              className="sc-gsFSjX hWksf"
              src={Object {}}
              style={
                Object {
                  "border": "5px solid #5E17EB",
                }
              }
            />
            <h3>
              Emily Buckley
            </h3>
             Senior Engineer
          </div>
          <div
            className="sc-aYaIB dQoBqn sc-kAycRU iekUvu"
          >
            <div>
              <a
                href="https://www.linkedin.com/in/emily-buckley-a91a7118a/"
                target="blank"
              >
                <svg
                  fill="#fff"
                  height="35"
                  viewBox="0 0 24 24"
                  width="35"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"
                  />
                </svg>
              </a>
            </div>
            <div>
              <a
                href="https://github.com/emilybuc"
                target="blank"
              >
                <svg
                  fill="#fff"
                  height="35"
                  viewBox="0 0 24 24"
                  width="35"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm2.218 18.616c-.354.069-.468-.149-.468-.336v-1.921c0-.653-.229-1.079-.481-1.296 1.56-.173 3.198-.765 3.198-3.454 0-.765-.273-1.389-.721-1.879.072-.177.312-.889-.069-1.853 0 0-.587-.188-1.923.717-.561-.154-1.159-.231-1.754-.234-.595.003-1.193.08-1.753.235-1.337-.905-1.925-.717-1.925-.717-.379.964-.14 1.676-.067 1.852-.448.49-.722 1.114-.722 1.879 0 2.682 1.634 3.282 3.189 3.459-.2.175-.381.483-.444.936-.4.179-1.413.488-2.037-.582 0 0-.37-.672-1.073-.722 0 0-.683-.009-.048.426 0 0 .46.215.777 1.024 0 0 .405 1.25 2.353.826v1.303c0 .185-.113.402-.462.337-2.782-.925-4.788-3.549-4.788-6.641 0-3.867 3.135-7 7-7s7 3.133 7 7c0 3.091-2.003 5.715-4.782 6.641z"
                  />
                </svg>
              </a>
            </div>
          </div>
          <p
            className="sc-imWZod fSMtJo"
          >
            Hey there! I'm Emily, ðŸ˜Š an experienced Software Engineer, boasting a solid 5+ years track record at LEGO and RS Components. My passion lies in tackling challenges head-on and crafting innovative solutions that make a difference. Beyond work, I love to unwind by immersing myself in the captivating world of board games, keeping fit at the gym, and playing with my cat Hugo ðŸ˜º
          </p>
        </div>
      </div>
      <div>
        <div
          className="sc-dcJtft eJDELm"
        >
          <div
            className="sc-iGgVNO ffHzaU"
          >
            <img
              alt="A staff photo of Greg"
              className="sc-gsFSjX hWksf"
              src={Object {}}
              style={
                Object {
                  "border": "5px solid #F1A1E7",
                }
              }
            />
            <h3>
              Gregory Carter
            </h3>
             Junior Engineer
          </div>
          <div
            className="sc-aYaIB dQoBqn sc-kAycRU iekUvu"
          >
            <div>
              <a
                href="https://www.linkedin.com/in/gregamo/"
                target="blank"
              >
                <svg
                  fill="#fff"
                  height="35"
                  viewBox="0 0 24 24"
                  width="35"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"
                  />
                </svg>
              </a>
            </div>
            <div>
              <a
                href="https://github.com/Gregamoo"
                target="blank"
              >
                <svg
                  fill="#fff"
                  height="35"
                  viewBox="0 0 24 24"
                  width="35"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm2.218 18.616c-.354.069-.468-.149-.468-.336v-1.921c0-.653-.229-1.079-.481-1.296 1.56-.173 3.198-.765 3.198-3.454 0-.765-.273-1.389-.721-1.879.072-.177.312-.889-.069-1.853 0 0-.587-.188-1.923.717-.561-.154-1.159-.231-1.754-.234-.595.003-1.193.08-1.753.235-1.337-.905-1.925-.717-1.925-.717-.379.964-.14 1.676-.067 1.852-.448.49-.722 1.114-.722 1.879 0 2.682 1.634 3.282 3.189 3.459-.2.175-.381.483-.444.936-.4.179-1.413.488-2.037-.582 0 0-.37-.672-1.073-.722 0 0-.683-.009-.048.426 0 0 .46.215.777 1.024 0 0 .405 1.25 2.353.826v1.303c0 .185-.113.402-.462.337-2.782-.925-4.788-3.549-4.788-6.641 0-3.867 3.135-7 7-7s7 3.133 7 7c0 3.091-2.003 5.715-4.782 6.641z"
                  />
                </svg>
              </a>
            </div>
          </div>
          <p
            className="sc-imWZod fSMtJo"
          >
            Hey, I'm Greg! ðŸ˜„ I trained as a video game designer but pursued a career in software engineering, where I thrive on solving complex technical challenges. My game design background fuels my creativity and problem-solving skills in the tech world. Outside of work, I love spending time with my daughter, attending live concerts, playing video games, and reading manga. I also enjoy going to the gym when I have time. This blend of technical expertise and diverse interests keeps me inspired and engaged in all aspects of my life.ðŸŒŸ
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
